### PR TITLE
Remove references to pilot and update doc link to new developer guide

### PR DESCRIPTION
**Pull request by knhage**
Make necessary updates regarding Pub/Sub API product GA.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,14 +1,17 @@
 # Getting Started with the Pub/Sub API
 
-- [About](#about)
-- [Resources](#resources)
+- [About Pub/Sub API](#about-pub/sub-api)
```
Comment (herinckc) [inline README.md:3]: I believe this should be `[About Pub/Sub API](#about-pubsub-api)` instead

File: README.md [hunk 2]
```diff
@@ -28,11 +31,15 @@ officially supported languages have well-supported Avro libraries:
...
 |Ruby|[AvroTurf](https://github.com/dasch/avro_turf)|
 
-## Resources
-
-**Documentation**
+## Documentation and Blog Posts
+- [Pub/Sub API Developer Guide](https://developer.salesforce.com/docs/platform/pub-sub-api/overview)
+- [Salesforce Architects Blog Post](https://medium.com/salesforce-architects/announcing-pub-sub-api-generally-available-3980c9eaf0b7)
 - [Salesforce Developers Blog Post](https://developer.salesforce.com/blogs/2021/07/pub-sub-api-building-event-driven-integrations-just-got-even-easier)
 - [Salesforce Developers Podcast Episode](https://developer.salesforce.com/podcast/2021/08/episode-93-pub-sub-api-with-emmett-chen-ran)
```
Comment (herinckc) [inline README.md:38]: Not directly related to this PR, but this link doesn't seem to work for me anymore, I'm just getting a message that says `Get notified of new episodes with the new Salesforce Developers Slack app.`. Would we want to either fix or remove this link as part of this PR?

File: README.md (hunk 1 above)
Comment (knhage) [reply README.md]: Fixed

File: README.md (hunk 2 above)
Comment (knhage) [reply README.md]: Remove last two links

File: README.md [hunk 3]
```diff
@@ -1,14 +1,17 @@
 # Getting Started with the Pub/Sub API
 
-- [About](#about)
-- [Resources](#resources)
+- [About Pub/Sub API](#about-pubsub-api)
+- [gRPC](#grpc)
+- [Documentation and Blog Posts](#documentation-and-blog-posts)
```
Comment (herinckc) [inline README.md:5]: Sorry, I missed this one. It looks like this should be `[Documentation and Blog Post](#documentation-and-blog-post)`

Comment (knhage) [reply README.md]: Good catch! Fixed.

**Review by herinckc — APPROVED** [review]

**Pull request merged**